### PR TITLE
lure: individualized links

### PR DESCRIPTION
**Pull request by arthyn**
This updates lure from the %groups perspective, changing %grouper and the web frontend to work with the new %reel flow. Fixes TLON-2517

PR Checklist
- [X] Includes changes to desk files
- [ ] Describes how you tested the PR locally (test ship vs livenet)
- [ ] If a new feature, includes automated tests
- [ ] Comments added anywhere logic may be confusing without context

### CODE REVIEW

**Comment by arthyn** [comment]
need to add changes to mobile as well

**Review by patosullivan — APPROVED** [review]
Frontend changes look good

File: desk/app/grouper.hoon [hunk 1]
```diff
@@ -36,39 +36,50 @@
...
     =/  baseurl  .^(cord %gx /(scot %p our.bowl)/reel/(scot %da now.bowl)/service/noun)
-    =/  target=ship  (slav %p i.t.path)
-    =/  group=cord  i.t.t.path
+    ::  it really is necessary to double-encode this, make sure we strip
+    ::  the leading slash before encoding
+    =/  end  (en-urlt:html (en-urlt:html +:(spud rest.pole)))
+    =/  url
+      ?.  =(baseurl 'https://tlon.network/lure/')
+        (crip "{(trip baseurl)}{end}")
+      (crip "https://tlon.network/v1/policies/lure/{end}")
```
Comment (Fang-) [inline desk/app/grouper.hoon:74]: Here and in urbit/landscape#280, it's surprising to me that the base url is expected to be given with a trailing slash, leading you into the annoying pattern of needing to strip the leading slash off the "actual" path... Unclear if that's worth correcting now, or punting to later.

(Also I maintain that double-encoding (or needing to encode at all for that matter) is a smell, but I get that it's finicky so we can just hold our noses for now.)

File: desk/app/grouper.hoon [hunk 2]
```diff
@@ -104,30 +126,48 @@
...
-      %kick       
+      %kick
     :_  this
     ~[(bite-subscribe bowl)]
     ::
       %fact
     =+  !<(=bite:reel q.cage.sign)
-    ~?  dev-mode  [bite (~(has in enabled-groups) token.bite)]
-    ?>  (~(has in enabled-groups) token.bite)
-    ?>  ?=([%bite-1 *] bite)
```
Comment (Fang-) [inline desk/app/grouper.hoon:115]: I don't see any changes here that stop producing `%bite-1` or start producing `%bite-2`. Am I missing something?

Also did `%bite-1` ever make it out into production-production? I thought that was your recent addition. If not, could be worth just updating the `%bite-1` case and calling it a day.

File: desk/app/grouper.hoon [hunk 3]
```diff
@@ -104,30 +126,48 @@
...
+    |.
+    ?~  group=(~(get by fields.metadata.bite) 'group')
+      ~|("no group field" !!)
+    =/  =flag:groups  (flag:dejs:gj s+u.group)
+    ~?  dev-mode  [bite (~(has in enabled-groups) q.flag)]
+    ?>  (~(has in enabled-groups) q.flag)
+    ~?  dev-mode  'inviting'
+    =/  =invite:groups  [flag joiner.bite]
+    =/  =path  /(scot %p our.bowl)/groups/(scot %da now.bowl)/groups/(scot %p p.flag)/[q.flag]/noun
     =+  .^(=group:groups %gx path)
```
Comment (Fang-) [inline desk/app/grouper.hoon:161]: Are you virtualizing here to save yourself if the scry crashes? Unfortunately, that won't save you. You'll want to do an explicit existence check.

Usually that'd be a `%gu` scry, but groups appears to implement a `%gx /exists/~some/group/noun`.

File: desk/app/grouper.hoon [hunk 4]
```diff
@@ -104,30 +126,48 @@
...
+      =/  wir=^wire  /dm/(scot %p joiner.bite)
+      =/  =dock  [our.bowl %chat]
+      =/  =id:c  [our now]:bowl
+      =/  =memo:c
+        [~[[%inline ~[[%ship joiner.bite] ' has joined the network']]] id]
+      =/  =action:dm:c
+        :-  joiner.bite
+        [id %add memo [%notice ~] ~]
+      =/  =cage  chat-dm-action+!>(`action:dm:c`action)
+      (welp caz ~[[%pass wir %agent dock %poke cage]])
```
Comment (Fang-) [inline desk/app/grouper.hoon:147]: For a single item, you can just `+snoc` it onto the end.

```suggestion
      (snoc caz [%pass wir %agent dock %poke cage])
```

**Review by Fang- — COMMENTED** [review]
Some comments. Primarily, you can't save yourself from scry crashes by virtualizing.

428f4e6 is a weird inclusion for this particular pr, and on its face I don't immediately like it because it (lightly) obscures the extent to which chat and channel structures are identical. I don't feel super strongly though.

File: desk/app/grouper.hoon (hunk 1 above)
Comment (arthyn) [reply desk/app/grouper.hoon:74]: yeah it's bad just going to hold nose until we make a simpler lure service

File: desk/app/grouper.hoon [hunk 5]
```diff
@@ -1,8 +1,8 @@
-/-  reel, groups
-/+  default-agent, verb, dbug
+/-  reel, groups, c=chat
+/+  gj=groups-json, default-agent, verb, dbug
 ::
 |%
-++  dev-mode  |
+++  dev-mode  &
```
[the remark this answers is not in the document]
Comment (arthyn) [reply desk/app/grouper.hoon:5]:
```suggestion
++  dev-mode  |
```

File: desk/app/grouper.hoon (hunk 2 above)
Comment (arthyn) [reply desk/app/grouper.hoon:115]: those changes went through on the landscape side here: https://github.com/tloncorp/landscape/blob/develop/desk/app/bait.hoon#L136-L150

File: desk/app/grouper.hoon (hunk 3 above)
Comment (arthyn) [reply desk/app/grouper.hoon:161]: Was virtualizing here for the early return if we don't have a group, but I guess I could just return empty list in both crash/assertions

File: desk/app/grouper.hoon [hunk 6]
```diff
@@ -138,7 +138,7 @@
       =/  wir=^wire  /dm/(scot %p joiner.bite)
       =/  =dock  [our.bowl %chat]
       =/  =id:c  [our now]:bowl
-      =/  =memo:c
+      =/  =memo:ch
```
[the remark this answers is not in the document]
Comment (arthyn) [reply desk/app/grouper.hoon:141]: 🤯

**Review by Fang- — APPROVED** [review]
Thanks for the changes, hoon code looks good!

**Pull request merged**